jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostEstimator.scala

```diff
@@ -107,18 +107,33 @@ class XGBoostEstimator private[spark](
     }
   }
 
+  private def ensureColumns(trainingSet: Dataset[_]): Dataset[_] = {
```

Comment:
Do you think we should force the user to always provide base margin? The purpose of this method to support the case when base margin was not provided (sadly, I haven't found a way to make `select` return None for missing column).